src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -303,6 +309,22 @@ private AsyncMethodHelper AsyncMethodHelper
             }
         }
 
+        private Assembly targetAssembly = null;
+        private bool tryToLoadAssembly = true;
+        private Assembly TargetAssembly
+        {
+            get
+            {
+                if (tryToLoadAssembly)
+                {
+                    targetAssembly = Assembly.LoadFrom(sourceAssembly);
```

Comment:
Here you are loading the assembly into the current AppDomain, whereas the original code does all reflection in a separate AppDomain.